api/importer/package_processor_test.go

```diff
@@ -1153,3 +1157,207 @@ func TestImportPackageProcessor_Process_FullManifestRendering(t *testing.T) {
+		Tasks: []*model.ManifestTask{
+			{
+				ID:      "",
+				Type:    "api",
+				Context: nil,
+				APITask: &model.APITask{
+					Action:      "keptn-api-v1-create-service",
+					PayloadFile: "",
+				},
+			},
+		},
+	}
+
+	expectedErrorMsg := "task id cannot be empty"
+
+	parserMock := &fake.ManifestParserMock{
+		ParseFunc: func(input io.Reader) (*model.ImportManifest, error) {
+			return new(model.ImportManifest), nil
+		},
+	}
+
+	taskExecutor := &fake.TaskExecutorMock{
+		ExecuteAPIFunc: func(ate model.APITaskExecution) (any, error) {
+			return nil, nil
+		},
+	}
+
+	importPackageMock := &fake.ImportPackageMock{
+		CloseFunc: func() error {
+			return nil
+		},
+		GetResourceFunc: func(resourceName string) (io.ReadCloser, error) {
+			return io.NopCloser(bytes.NewReader([]byte{})), nil
+		},
+	}
+
+	stageRetriever := &fake.MockStageRetriever{}
+	sut := NewImportPackageProcessor(parserMock, taskExecutor, stageRetriever)
+
+	err := sut.validateManifest(manifest, importPackageMock)
+
+	assert.EqualErrorf(t, err, expectedErrorMsg, "Error should be: %v, got: %v", expectedErrorMsg, err)
+}
+
+func TestPackageValidationIdAlphaNumeric(t *testing.T) {
+	manifests := []model.ImportManifest{
+		{
+			ApiVersion: "v1beta1",
+			Tasks: []*model.ManifestTask{
+				{
+					ID:      "valid_id",
+					Type:    "api",
+					Context: nil,
+					APITask: &model.APITask{
+						Action:      "keptn-api-v1-create-service",
+						PayloadFile: "",
+					},
+				},
+			},
+		},
+		{
+			ApiVersion: "v1beta1",
+			Tasks: []*model.ManifestTask{
+				{
+					ID:      "invalid-id",
+					Type:    "api",
+					Context: nil,
+					APITask: &model.APITask{
+						Action:      "keptn-api-v1-create-service",
+						PayloadFile: "",
+					},
+				},
+			},
+		},
+		{
+			ApiVersion: "v1beta1",
+			Tasks: []*model.ManifestTask{
+				{
+					ID:      "invalid~char#",
+					Type:    "api",
+					Context: nil,
+					APITask: &model.APITask{
+						Action:      "keptn-api-v1-create-service",
+						PayloadFile: "",
+					},
+				},
+			},
+		},
+	}
+
+	parserMock := &fake.ManifestParserMock{
+		ParseFunc: func(input io.Reader) (*model.ImportManifest, error) {
+			return new(model.ImportManifest), nil
+		},
+	}
+
+	taskExecutor := &fake.TaskExecutorMock{
+		ExecuteAPIFunc: func(ate model.APITaskExecution) (any, error) {
+			return nil, nil
+		},
+	}
+
+	importPackageMock := &fake.ImportPackageMock{
+		CloseFunc: func() error {
+			return nil
+		},
+		GetResourceFunc: func(resourceName string) (io.ReadCloser, error) {
+			return io.NopCloser(bytes.NewReader([]byte{})), nil
+		},
+	}
+
+	stageRetriever := &fake.MockStageRetriever{}
+	sut := NewImportPackageProcessor(parserMock, taskExecutor, stageRetriever)
+
+	tests := []struct {
+		name               string
+		manifest           model.ImportManifest
+		expectedErrMessage string
+		expectError        bool
+	}{
+		{
+			name:               "valid id",
+			manifest:           manifests[0],
+			expectedErrMessage: "<nil>",
```

Comment:
To avoid confusion when reading the expectedErrMessage can be left at its zero-value since it's irrelevant in this testcase (expectError is false)
```suggestion
```